woocommerce/woocommerce-gutenberg-products-block
# Product Gallery > Pager block: Add support to interactivity API

## What
This PR adds support to the Interactivity API on the Paget block, allowing shoppers to navigate between the gallery images without having to reload the page.

Fixes #10630

## Why

This is part of our effort to implement the new Product Gallery block.



## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. From your WordPress dashboard, go to Appearance > Themes. Make sure you have a block-based theme installed and activated. If not, you can install one from the Add New option. Block-based themes include "Twenty-twenty Two," "Twenty-twenty Three," etc.
2. On the left-hand side menu, click on Appearance > Editor > Templates
3. Find and select the 'Single Product' template from the list.
4. When the Classic Product Template renders, click on Transform into Blocks. This will transform the Classic template in a block template if you haven't done it before.
5. In the block library, search for the 'Product Gallery' block. Click on it to add the block to the template.
6. Check that the Pager block is displayed on the Editor, and its display mode is set to "Dots":
![CleanShot 2023-09-12 at 19 06 04](https://github.com/woocommerce/woocommerce-blocks/assets/20469356/e89fbf6d-a935-41b2-8db7-93be45c930b5)
7. Click on save and visit a product page that contains multiple gallery images, make sure the block is correctly being displayed and when selecting a different page, the Large Image changes to reflect the selected image.
8. Change the display mode to "Digits" (follow step 6) and repeat step 7.
9. Change the display mode to "Off" (follow step 6), click on Save and visit a product page that contains multiple gallery images, make sure the block is not displayed.


## Screenshots or screencast



| Display mode: Dots | Display mode: Digits | Display mode: Off |
| ------ | ----- | ----- |
| ![CleanShot 2023-09-12 at 19 11 44](https://github.com/woocommerce/woocommerce-blocks/assets/20469356/aa5f4b6d-adf5-4ec4-bf45-8772f537530a) | ![CleanShot 2023-09-12 at 19 11 15](https://github.com/woocommerce/woocommerce-blocks/assets/20469356/083b87fa-2da4-4eec-9c69-8ed1687d4d86) | ![CleanShot 2023-09-12 at 19 12 34](https://github.com/woocommerce/woocommerce-blocks/assets/20469356/3f68e4a3-b432-4e27-bb2e-c31059f25a8d) |

## WooCommerce Visibility


Required:
